api/tests/phones_views_tests.py

```diff
@@ -899,6 +902,111 @@ def test_inbound_sms_reply(phone_user, mocked_twilio_client):
+    real_phone = _make_real_phone(phone_user, verified=True)
+    relay_number = _make_relay_number(phone_user, enabled=True)
+    inbound_contact = InboundContact.objects.create(
+        relay_number=relay_number,
+        inbound_number="+15556660000",
+        last_inbound_type="text",
+        last_text_date=timezone.now(),
+    )
+    # inbound contact then calls
+    _check_and_update_contact(inbound_contact, "calls", relay_number)
+    assert inbound_contact.last_inbound_type == "call"
+    assert inbound_contact.last_call_date is not None
+    mocked_twilio_client.reset_mock()
+
+    client = APIClient()
+    path = "/api/v1/inbound_sms"
+    data = {"From": real_phone.number, "To": relay_number.number, "Body": "test reply"}
+    response = client.post(path, data, HTTP_X_TWILIO_SIGNATURE="valid")
+
+    assert response.status_code == 200
+    mocked_twilio_client.messages.create.assert_called_once()
+    call_kwargs = mocked_twilio_client.messages.create.call_args.kwargs
+    assert call_kwargs["to"] == inbound_contact.inbound_number
+    assert call_kwargs["from_"] == relay_number.number
+    assert call_kwargs["body"] == "test reply"
+    relay_number.refresh_from_db()
+    assert relay_number.texts_forwarded == 1
+
+
+def test_inbound_sms_reply_to_caller(
+    phone_user: User, mocked_twilio_client: Client
+) -> None:
+    """MPP-2581: A later contact that texts then calls is still the last texter."""
+    real_phone = _make_real_phone(phone_user, verified=True)
+    relay_number = _make_relay_number(phone_user, enabled=True)
+    inbound_contact1 = InboundContact.objects.create(
+        relay_number=relay_number,
+        inbound_number="+13015550000",
+        last_inbound_type="text",
+        last_inbound_date=timezone.now() - timedelta(seconds=3600),
+        last_text_date=timezone.now() - timedelta(seconds=3600),
+    )
+    inbound_contact2 = InboundContact.objects.create(
+        relay_number=relay_number,
+        inbound_number="+13025550001",
+        last_inbound_type="text",
+        last_inbound_date=timezone.now() - timedelta(seconds=1800),
+        last_text_date=timezone.now() - timedelta(seconds=1800),
+    )
+    # second inbound contact then calls
+    _check_and_update_contact(inbound_contact2, "calls", relay_number)
+    assert inbound_contact2.last_inbound_type == "call"
+    assert inbound_contact2.last_call_date is not None
+    mocked_twilio_client.reset_mock()
+
+    client = APIClient()
+    path = "/api/v1/inbound_sms"
+    data = {"From": real_phone.number, "To": relay_number.number, "Body": "test reply"}
+    response = client.post(path, data, HTTP_X_TWILIO_SIGNATURE="valid")
+
+    assert response.status_code == 200
+    mocked_twilio_client.messages.create.assert_called_once()
+    call_kwargs = mocked_twilio_client.messages.create.call_args.kwargs
+    assert call_kwargs["to"] != inbound_contact1.inbound_number
+    assert call_kwargs["to"] == inbound_contact2.inbound_number
```

Comment:
question (blocking) Is this test right? This test seems like it's saying that a text reply should go to the number that sent the last *call* ? Is that really what we want?

Reply:
This is the sequence from the user's perspective:

1. Contact 1 texts
2. Contact 2 texts
3. Contact 2 calls

I think the text reply should go to the last texter, Contact 2, even though they then called. Before this PR, it goes to Contact 1.

Reply:
Makes sense, but I don't see those steps 1 & 2 in this test?